lib/src/request_processor.cc

```diff
@@ -915,38 +923,48 @@ void RequestProcessor::logLoadedModules() const
 }
 
 //
-// Spool directory purge task (private interface)
+// Resource purge task (private interface)
 //
 
-void RequestProcessor::spoolDirPurgeTask()
+static unsigned int minutes_gcd(std::vector<std::shared_ptr<Util::Purgeable>> purgeables)
 {
-    auto num_minutes = Timestamp::getMinutes(spool_dir_purge_ttl_);
-    num_minutes *= 1.2;
-    // TODO(ale): deal with locale & plural (PCP-257)
-    if (num_minutes == 1) {
-        LOG_INFO("Starting the task for purging the spool directory every {1} "
-                 "minute; thread id {2}",
-                 num_minutes, pcp_util::this_thread::get_id());
+    assert(!purgeables.empty());
+    auto init = Timestamp::getMinutes(purgeables.front()->get_ttl());
+    if (purgeables.size() > 1) {
+        return std::accumulate(purgeables.begin()+1, purgeables.end(), init,
+            [](unsigned int result, std::shared_ptr<Util::Purgeable> &p) {
+                return boost::math::gcd(result, Timestamp::getMinutes(p->get_ttl()));
+            });
     } else {
-        LOG_INFO("Starting the task for purging the spool directory every {1} "
-                 "minutes; thread id {2}",
-                 num_minutes, pcp_util::this_thread::get_id());
+        return init;
     }
+}
+
+void RequestProcessor::purgeTask()
+{
+    // Use min of 1h and gcd of purgeable TTLs.
+    auto num_minutes = std::min(60u, minutes_gcd(purgeables_));
```

Comment:
Using gcd might be overly aggressive here. Poor choice of TTLs could result in checking every minute, and we probably don't need to be that precise.